### ISSUE TITLE
Issues with image, packages, theme, cpu speed and gpu on Elementary OS

### ISSUE THREAD

**Issue by Traverse** [dylanaraps/fetch#26]
I've encountered several discrepancies compared to `screenfetch` on Elementary OS. _(All dependencies are installed)_

![screenshot from 2016-01-23 14 07 53](https://cloud.githubusercontent.com/assets/7240988/12530310/d0c625b4-c1db-11e5-80f3-cbfa4ecaabd2.png)
![screenshot from 2016-01-23 14 03 48](https://cloud.githubusercontent.com/assets/7240988/12530309/d0c5e7ac-c1db-11e5-9314-3a07e52f8471.png)
![screenshot from 2016-01-23 14 12 02](https://cloud.githubusercontent.com/assets/7240988/12530311/d0c62c30-c1db-11e5-8b04-12aa820e82f5.png)

So to sum it up, background image isn't being displayed, packages aren't being displayed, the WM theme isn't being displayed, the CPU speeds don't match (should be 1.9GHz), and it only shows the integrated graphics.

Packages is easily fixed by including `"elementary"*` with `"Ubuntu"* | "Mint"* | "Debian"* | "Kali Linux"* | "Deepin Linux"*`.

For the other parts I don't have any quick fixes sadly.

**Comment by dylanaraps** [comment]
Thanks for the issue!

- Package count has been fixed.

- CPU speed should be fixed.

- For GTK theme / Icons the function looks in `$HOME/.gtkrc-2.0` and `$HOME/.config/gtk-3.0/settings.ini` are these files not in those locations on elementary?

Also thanks for testing GPU support, I've added a rule to replace `SeymourRadeon` with `Radeon` and if `gpu_shorthand="on"` the word `Series` will also be removed from the output.

I'm going to install Elementary in a virtual machine to see what's going on with the GTK theme stuff.

**Comment by Traverse** [comment]
I had `feh` installed but it wasn't showing my wallpaper, I might be doing something wrong though not 100% sure. I'm sadly on a windows machine right now so can't really test that and GTK theme / icons location.

If I have some time to look into it before you do I'll share the findings here.

**Comment by dylanaraps** [comment]
We might be able to add support for elementary's wallpaper setter, all we need to do is find out where the current wallpaper is stored. 

My internet is terrible so it'll take me another hour or so to download the iso.

**Comment by dylanaraps** [comment]
Everything but the wallpaper issue has been fixed.

**Comment by dylanaraps** [comment]
All issues have been fixed.

I'll close the issue once you've tested for yourself.

**Comment by Traverse** [comment]
Will probably have some time to test after I've slept so expect an update in ~8 hours.

**Comment by Traverse** [comment]
For me the GTK  theme / icons still don't show up, images still don't display and it still shows the wrong CPU speed (current not max). Not sure if it depends on my install, can try a clean install on a VM in a bit.

![screenshot from 2016-01-24 13 11 10](https://cloud.githubusercontent.com/assets/7240988/12536145/26c39300-c29c-11e5-8854-02388929e71c.png)

**Comment by dylanaraps** [comment]
I tested on a fresh VM with elementary and it worked fine. Weird.

I'm getting cpu speed from `/proc/cpuinfo`, this is the command that I'm using. What does it output on your system?

```sh
awk -F ': ' '/cpu MHz/ {printf $2; exit}' /proc/cpuinfo
```

I'm getting the current gtk theme from `$HOME/.gtkrc-2.0` and `$HOME/.config/gtk-3.0/settings.ini` first and then falling back to `/etc/gtk-2.0/gtkrc` and `/etc/gtk-3.0/settings.ini`. Do these files exist on your system?

Do images appear in ranger and w3m? If not then it's an issue with elementary's terminal emulator and not fetch.

I'll also fix that spacing issue with your cpu speed.

**Comment by Traverse** [comment]
Just tried on my VM with Elementary and concluded the same about the images. 

As for the CPU speed I'm thinking it actually depends on the CPU because on the VM it kept giving the correct values repeatedly and on my real machine it kept giving the current CPU speed.

The GTK theme / icon thing might have to do with the way you switch themes on Elementary, you have to use Elementary-Tweaks to switch the themes. For example on my normal install it shows the theme as `none` and on the VM install it shows a theme but not the one currently being used.

**Comment by dylanaraps** [comment]
I see.

Is `elementary-tweaks` installed by default? I don't have it in in my VM and I'd like to find a solution that works for 100% of elementary users instead of only those with `elementary-tweaks` installed.

Seeing as the images and cpu speed issues are out of my hands, is the only issue you're now having the GTK theme/icons not showing the correct value?

**Comment by dylanaraps** [comment]
Do any of these command output the correct theme value?

```sh
gconftool-2 -g /desktop/gnome/interface/gtk_theme

gsettings get org.gnome.desktop.interface gtk-theme

```

**Comment by dylanaraps** [comment]
GTK theme detection should be fixed now.

**Comment by Traverse** [comment]
GTK theme detection works now. Only way I can imagine on how to fix the CPU part is copying what `screenfetch` is doing which might not be the most ideal solution.

**Comment by dylanaraps** [comment]
I added some more speed types that you can choose from. You can change the cpu speed type using `--speed_type` or by changing `$speed_type` inside the script. Here are the values:

- `current`: Print the current cpu speed using cpufreq
- `min`: Print the min cpu speed using cpufreq
- `max`: Print the max cpu speed using cpufreq
- `bios`: Print the bios max cpu speed. **This is the one that will print your correct cpu speed**
- `scaling_current`: Print the current cpu scaling speed
- `scaling_min`: Print the min cpu scaling speed
- `scaling_max`: Print the max cpu scaling speed

Try them all out but I'm fairly certain that `bios` is the value you'll want to use.

**Comment by Traverse** [comment]
I'll try it

**Comment by dylanaraps** [comment]
Is everything working fine now?

**Comment by Traverse** [comment]
Yep everything working as expected now, thanks a bunch for looking into this!

**Comment by dylanaraps** [comment]
No problem!
